mipengine/controller/common/algorithms_specifications.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import os
+import typing
+from dataclasses import dataclass
+from typing import List, Dict, Optional, Any
+
+from dataclasses_json import dataclass_json
+
+from mipengine.controller.common.utils import Singleton
+
+# TODO How can we read all algorithm.json files without relative paths?
+RELATIVE_ALGORITHMS_PATH = "./mipengine/algorithms"
```

Comment:
If I understand correctly, the problem is that you want to look for json files which cannot be imported in python. Since they are in the folder algorithms you can do the following:
```
from mipengine import algorithms
path = algorithms.__file__
```
This is actually the package's `__init__.py` path.

Then I suggest using `pathlib` instead of `os.path`. It much more intuitive.
E.g.:
```
from pathlib import Path
from mipengine import algorithms
path = Path(algorithms.__file__).parent
jsons = [file for file in path.glob('*.json')]
```

Reply:
Nice, thanks!